### PR TITLE
move `block_ddp_sync_behaviour` to utilities

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Moves the context manager `block_ddp_sync_behaviour` in training loop to utilities. In addition:

- renamed to block_parallel_sync_behavior because it is not ddp specific, applies generally to all parallel plugins
- renamed "behaviour" to "behavior" since majority of our code base and docs uses american english
- renamed the argument "should_block_sync" to just "block" for brevity.
- updated docstring
- made it protected to avoid reviewer commenting that it's public

Part of #9191 
Related (but not blocking each other): #9101

### Does your PR introduce any breaking changes? If yes, please list them.

While this method was part of the public API, nobody can really effectively use it externally, so we don't consider this a breaking change at all.

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

### CODE REVIEW

**Review by tchaton — APPROVED** [review]
Neat !

**Review by carmocca — APPROVED** [review]

File: pytorch_lightning/loops/utilities.py [hunk 1]
```diff
@@ -113,3 +114,22 @@ def _prepare_dataloader_iter(data_fetcher: AbstractDataFetcher, batch_idx: int)
     else:
         dataloader_iter = iter(data_fetcher)
     return dataloader_iter
+
+
+@contextmanager
+def _block_parallel_sync_behavior(trainer: "pl.Trainer", block: bool = True) -> Generator[None, None, None]:
+    """Blocks synchronization in :class:`~pytorch_lightning.plugins.training_type.parallel.ParallelPlugin`.
```
Comment (kaushikb11) [inline pytorch_lightning/loops/utilities.py:121]:
```suggestion
    """
    Blocks synchronization in :class:`~pytorch_lightning.plugins.training_type.parallel.ParallelPlugin`.
```
Could start on the next line?

**Review by kaushikb11 — APPROVED** [review]

File: pytorch_lightning/loops/utilities.py (hunk 1 above)
Comment (awaelchli) [reply pytorch_lightning/loops/utilities.py:121]: ah ok, I can do that
I thought since we have blacken-docs this was the new standard
